library/core/src/ptr/mod.rs

```diff
@@ -1451,9 +1451,11 @@ fnptr_impls_args! { A, B, C, D, E, F, G, H, I, J, K, L }
 /// as all other references. This macro can create a raw pointer *without* creating
 /// a reference first.
 ///
-/// Note, however, that the `expr` in `addr_of!(expr)` is still subject to all
-/// the usual rules. In particular, `addr_of!(*ptr::null())` is Undefined
-/// Behavior because it dereferences a null pointer.
+/// Note, however, that the `expr` in `addr_of!(expr)` is still subject to other rules:
+/// * A [dangling] pointer cannot be dereferenced. In particular, `addr_of!(*ptr::null_mut())`
+///   is undefined behavior because it dereferences a null pointer.
+/// * An unaligned pointer cannot be dereferenced. For example, `addr_of!((*an_unaligned_ptr).field)`
```

Comment:
does unaligned deserve a link to a better description / example or will everyone that reads this have that context already?

Reply:
Good call, I've added a link there to the [Type layout](https://doc.rust-lang.org/reference/type-layout.html) reference page.